backend/src/services/consumer-secret/consumer-secret-service.ts

```diff
@@ -0,0 +1,143 @@
+import { TPermissionServiceFactory } from "@app/ee/services/permission/permission-service";
+import { UnauthorizedError } from "@app/lib/errors";
+import { TConsumerSecretAttributeDALFactory } from "@app/services/consumer-secret-attribute/consumer-secret-attribute-dal";
+
+import { TKmsServiceFactory } from "../kms/kms-service";
+import { KmsDataKey } from "../kms/kms-types";
+import { TConsumerSecretDALFactory } from "./consumer-secret-dal";
+import {
+  TCreateConsumerSecretDTO,
+  TDeleteConsumerSecretDTO,
+  TGetConsumerSecretsDTO,
+  TUpdateConsumerSecretDTO
+} from "./consumer-secret-types";
+
+type TConsumerSecretServiceFactoryDep = {
+  consumerSecretDAL: TConsumerSecretDALFactory;
+  consumerSecretAttributeDAL: TConsumerSecretAttributeDALFactory;
+  kmsService: TKmsServiceFactory;
+  permissionService: Pick<TPermissionServiceFactory, "getUserOrgPermission">;
+};
+
+export type TConsumerSecretServiceFactory = ReturnType<typeof consumerSecretServiceFactory>;
+
+// todo: implement permissions
+export const consumerSecretServiceFactory = ({
+  consumerSecretDAL,
+  consumerSecretAttributeDAL,
+  kmsService
+}: TConsumerSecretServiceFactoryDep) => {
+  const createConsumerSecret = async ({
+    actorId,
+    actorOrgId,
+    actorAuthMethod,
+    attributes,
+    ...input
+  }: TCreateConsumerSecretDTO) => {
+    if (!actorOrgId) throw new UnauthorizedError({ message: "No organization ID provided in request" });
+
+    const consumerSecret = await consumerSecretDAL.transaction(async (tx) => {
+      const newSecret = await consumerSecretDAL.create(
+        {
+          name: input.name,
+          type: input.type,
+          userId: actorId,
+          orgId: actorOrgId
+        },
+        tx
+      );
+
+      if (attributes && attributes.length > 0) {
+        const { encryptor: secretManagerEncryptor } = await kmsService.createCipherPairWithDataKey(
+          {
+            type: KmsDataKey.Organization,
+            orgId: actorOrgId
+          },
+          tx
+        );
```

Comment:
Self-review: I think we might be able to make things a little more secure by implementing user_org key pairs. I am just not entirely sure it is warranted—would love feedback on this